### PR TITLE
ref: fix more typing issues

### PR DESCRIPTION
**Pull request by asottile-sentry**

### CODE REVIEW

File: tests/sentry/ingest/billing_metrics_consumer/test_billing_metrics_consumer_kafka.py [hunk 1]
```diff
@@ -22,15 +24,16 @@ def test_outcomes_consumed(track_outcome):
...
+    buckets: list[MetricsBucket] = [
         {  # Counter metric with wrong ID will not generate an outcome
             "metric_id": 123,
             "type": "c",
             "org_id": 1,
             "project_id": 2,
             "timestamp": 123,
             "value": 123.4,
-            "tags": {},
+            "tags": empty_tags,
```
Comment (iker-barriocanal) [inline tests/sentry/ingest/billing_metrics_consumer/test_billing_metrics_consumer_kafka.py:36]: Not sure if the linter should fail here... 🤔

**Review by iker-barriocanal — APPROVED** [review]

File: src/sentry/integrations/slack/unfurl/discover.py [hunk 2]
```diff
@@ -101,10 +101,10 @@ def is_aggregate(field: str) -> bool:
 
 
 def unfurl_discover(
-    data: HttpRequest,
```
Comment (armenzg) [inline src/sentry/integrations/slack/unfurl/discover.py:104]: Am I loosing it? I don't see `data` being used in this function.

**Review by armenzg — APPROVED** [review]
I've looked at integrations. Thank you @asottile-sentry !

File: tests/sentry/ingest/billing_metrics_consumer/test_billing_metrics_consumer_kafka.py (hunk 1 above)
Comment (asottile-sentry) [reply tests/sentry/ingest/billing_metrics_consumer/test_billing_metrics_consumer_kafka.py:36]: this is a microbug with mypy -- it doesn't know that `{}` (`dict[<nothing>, <nothing>]`) solves the variance for `Mapping[K, V]`

**Pull request merged**